FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.DataWriter/Azure/AzureBlobContainerClient.cs

```diff
@@ -155,19 +164,27 @@ private DataLakeFileSystemClient DataLakeFileSystemClient
             set => _dataLakeFileSystemClient = value;
         }
 
-        private void InitializeBlobContainerClient(BlobContainerClient blobContainerClient)
+        private bool InitializeBlobContainerClient(BlobContainerClient blobContainerClient)
         {
             try
             {
                 blobContainerClient.CreateIfNotExists();
                 _logger.LogInformation($"Create container {blobContainerClient.Name} successfully.");
             }
+            catch (RequestFailedException ex) when (IsAuthenticationError(ex))
+            {
+                _diagnosticLogger.LogError($"Create container {blobContainerClient.Name} failed. Reason: {ex.Message}");
+                _logger.LogInformation(ex, $"Failed to create container {blobContainerClient.Name} due to authentication issue.");
+                return true;
+            }
             catch (Exception ex)
             {
                 _diagnosticLogger.LogError($"Create container {blobContainerClient.Name} failed. Reason: {ex.Message}");
-                _logger.LogInformation(ex, $"Create container {blobContainerClient.Name} failed. Reason: {ex.Message}");
-                throw new AzureBlobOperationFailedException($"Create container {blobContainerClient.Name} failed.", ex);
+                _logger.LogError(ex, $"Failed to create container {blobContainerClient.Name}.");
+                throw new AzureBlobOperationFailedException($"Failed to create container {blobContainerClient.Name}.", ex);
             }
+
+            return false;
```

Comment:
Feel a little strange for returning false if success, maybe return true and change outer logic?

Reply:
Ignore